packages/core/src/operator/getConfirmDataAndHash.ts

```diff
@@ -0,0 +1,33 @@
+import { 
+    getConfirmDatafromGraph, 
+    getSubgraphHealthStatus, 
+    getMultipleChallengeConfirmData,
+} from "../index.js";
+
+/**
+ * Fetches the confirmData and confirmHash of a given array of assertion Ids.
+ * @param assertionIds - The array of assertion IDs.
+ * @returns The confirmData and confirmHashS.
+ */
+export async function getConfirmDataAndHash(
+    assertionIds: number[],
+): Promise<{confirmData: string[], confirmHash: string}> {
+
+    //declare return variables
+    let confirmData: string[];
+    let confirmHash: string;
+    
+    //determine whether to hit subgraph or rpc
+    const graphStatus = await getSubgraphHealthStatus();
+    if (graphStatus.healthy) {
```

Comment:
This function will be called in the operator runtime, we will already have called the subgraph health status, is there no way to pass that in here to not have to call again. 

Calling the healthstatus counts as 1 subgraph query :(

Reply:
Added a boolean parameter `isSubgraphHealthy` to the function to decide whether to hit the subgraph or the RPC.